src/explore-education-statistics-frontend/src/services/azurePublicationService.ts

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable no-restricted-syntax */
+import { env } from 'process';
+import { PublicationListSummary } from '@common/services/publicationService';
+import { PaginatedList } from '@common/services/types/pagination';
+import {
+  AzureKeyCredential,
+  FacetResult,
+  SearchClient,
+} from '@azure/search-documents';
+import { ReleaseType } from '@common/services/types/releaseType';
+import { SortDirection } from '@common/services/types/sort';
+
+export interface AzurePublicationSearchResult {
+  '@search.score': string;
+  content: string;
+  releaseSlug: string;
+  releaseType: string;
+  releaseVersionId: string;
+  publicationId: string;
+  publicationSlug: string;
+  published: Date;
+  releaseId: string;
+  summary: string;
+  themeId: string;
+  themeTitle: string;
+  title: string;
+  typeBoost: string;
+  metadata_storage_last_modified: Date;
+  metadata_storage_name: string;
+}
+
+export interface PaginatedListWithAzureFacets<T> extends PaginatedList<T> {
+  facets: {
+    [propertyName: string]: FacetResult[];
+  };
+}
+
+export type AzurePublicationOrderByParam =
+  | 'published asc'
+  | 'published desc'
+  | 'title asc'
+  | undefined;
+
+export interface AzurePublicationListRequest {
+  filter?: string;
+  page?: number;
+  pageSize?: number;
+  search?: string;
+  orderBy?: AzurePublicationOrderByParam;
+  sortDirection?: SortDirection;
+}
+
+const { AZURE_SEARCH_ENDPOINT, AZURE_SEARCH_INDEX, AZURE_SEARCH_QUERY_KEY } =
+  env;
+
+const azureSearchClient = new SearchClient<AzurePublicationSearchResult>(
+  AZURE_SEARCH_ENDPOINT || '',
```

Comment:
Looks like we also need to add this endpoint URL to `cspConnectSrc` in `server.js` to avoid the browser blocking the request because the URL is not in the site's content security policy.